distributed/cli/utils.py

```diff
@@ -1,30 +1,28 @@
-from tornado.ioloop import IOLoop
+from __future__ import annotations
 
+import asyncio
+import logging
+import signal
+from typing import Any
 
-def install_signal_handlers(loop=None, cleanup=None):
-    """
-    Install global signal handlers to halt the Tornado IOLoop in case of
-    a SIGINT or SIGTERM.  *cleanup* is an optional callback called,
-    before the loop stops, with a single signal number argument.
-    """
-    import signal
+logger = logging.getLogger(__name__)
 
-    loop = loop or IOLoop.current()
 
-    old_handlers = {}
+async def wait_for_signals(signals: list[signal.Signals]) -> None:
+    """Wait for the passed signals by setting global signal handlers"""
+    loop = asyncio.get_running_loop()
+    event = asyncio.Event()
 
-    def handle_signal(sig, frame):
-        async def cleanup_and_stop():
-            try:
-                if cleanup is not None:
-                    await cleanup(sig)
-            finally:
-                loop.stop()
+    old_handlers: dict[int, Any] = {}
 
-        loop.add_callback_from_signal(cleanup_and_stop)
-        # Restore old signal handler to allow for a quicker exit
+    def handle_signal(signum, frame):
+        # Restore old signal handler to allow for quicker exit
         # if the user sends the signal again.
-        signal.signal(sig, old_handlers[sig])
+        signal.signal(signum, old_handlers[signum])
+        logger.info("Received signal %d", signum)
```

Comment:
I love that we no longer get the timeout error. Slight UX improvement: Print the signal name, not the number (or both), e.g.

```suggestion

        logger.info("Received signal %s (%d)", signal.Signals(signum).name, signum)
```

prints

```
^C2022-05-17 14:09:26,685 - distributed.cli.utils - INFO - Received signal SIGINT (2)

```